sdk/schemaregistry/azure-schemaregistry-avroserializer/tests/perf_tests/serialize_deserialize_cache.py

```diff
@@ -0,0 +1,153 @@
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import os
+import time
+import json
+import sys
+
+from azure.identity import ClientSecretCredential
+from azure.schemaregistry import SchemaRegistryClient
+from azure.schemaregistry.serializer.avroserializer import SchemaRegistryAvroSerializer
+
+TENANT_ID=os.environ['SCHEMA_REGISTRY_AZURE_TENANT_ID']
+CLIENT_ID=os.environ['SCHEMA_REGISTRY_AZURE_CLIENT_ID']
+CLIENT_SECRET=os.environ['SCHEMA_REGISTRY_AZURE_CLIENT_SECRET']
+
+SCHEMA_REGISTRY_ENDPOINT=os.environ['SCHEMA_REGISTRY_ENDPOINT']
+GROUP_NAME=os.environ['SCHEMA_REGISTRY_GROUP']
+
+
+token_credential = ClientSecretCredential(
+    tenant_id=TENANT_ID,
+    client_id=CLIENT_ID,
+    client_secret=CLIENT_SECRET
+)
+
+cached_call_count = 100
+num_long_schema_entries = 10000
+
+SCHEMA_STRING_SHORT = """
+{"namespace": "example.short.avro",
+ "type": "record",
+ "name": "User",
+ "fields": [
+     {"name": "name", "type": "string"}
+ ]
+}"""
+SCHEMA_DICT_LONG = {"namespace": "example.med.avro",
+ "type": "record",
+ "name": "User",
+ "fields": [
+     {"name": "name", "type": "string"},
+     {"name": "favorite_color", "type": "string"}
+ ]
+}
+for i in range(num_long_schema_entries):
+    SCHEMA_DICT_LONG["fields"].append({"name": f"num{i}", "type": "int"})
+SCHEMA_STRING_LONG = json.dumps(SCHEMA_DICT_LONG)
+
+
+def serialize(serializer, dict_data_a, dict_data_b, schema_str):
+
+    # Schema would be automatically registered into Schema Registry and cached locally.
+    st1 = time.time()
+    payload_ben = serializer.serialize(dict_data_b, schema_str)
+    et1 = time.time()
+    # The second call won't trigger a service call.
+    st2 = time.time()
+    for _ in range(cached_call_count):
+        payload_alice = serializer.serialize(dict_data_a, schema_str)
+    et2 = time.time()
+
+    print("time to serialize one value with uncached schema: {}".format(et1-st1))
+    print("avg time to serialize one value with cached schema: {}".format((et2-st2)/cached_call_count))
+    return [payload_ben, payload_alice]
```

Comment:
you could use [timeit](https://docs.python.org/3/library/timeit.html) to run perf analysis for small snippets.

besides, I think we could make those be part of the unit test. So that we don't need to run the scripts manually to get the results. CI/Livetest will run the scripts for us.
with that being said, it requires us to change the scripts a bit instead of just printing out the perf. We can get some baseline perf results, do assertion (compare perf) to ensure there's no performance regression.
It doesn't need to happen in this PR/this sprint, we could log an issue to track it.
If perf is something we care before GA, then we do it before GA, otherwise we could address it in MQ.

to merge this PR, let's try using timeit first 😉

Reply:
sounds good!